Project.toml

```diff
@@ -4,8 +4,6 @@ authors = ["Markus Hauru <markus@mhauru.org>", "Jutho Haegeman <jutho.haegeman@u
 version = "0.1.0"
 
 [deps]
-ArgParse = "c7e460c6-2fb9-53a9-8c5b-16f535851c63"
-JLD2 = "033835bb-8acc-5ee8-8aae-3f567f8a3819"
```

Comment:
The stuff in `demo` actually depends on JLD2. It's not the best solution to have the whole main project have JLD2 as a dependency when it's only the demo stuff that uses it, but I'm not sure what the best way to solve this.

Reply:
Oh I see; you could have a separate project file in `demo`?

Reply:
Yeah, that would probably be the way to go. I've forgotten how exactly projects and dependency management in Julia works, if that would introduce extra headaches of some kind. I'm fine merging this anyway, fixing the demo stuff later.

Reply:
I can try to add a project file to the demo; feel free to hold of merging until I do.

Reply:
Sure. Let me know though if this PR is blocking something else, we can also come back to the demo part later, it's secondary to the rest.